dpnp/dpnp_iface_logic.py

```diff
@@ -923,6 +1012,120 @@ def isposinf(x, out=None):
     return dpnp.logical_and(is_inf, signbit, out=out)
 
 
+def isreal(x):
+    """
+    Returns a bool array, where ``True`` if input element is real.
+
+    If element has complex type with zero imaginary part, the return value
+    for that element is ``True``.
+
+    For full documentation refer to :obj:`numpy.isreal`.
+
+    Parameters
+    ----------
+    x : {dpnp.ndarray, usm_ndarray}
+        Input array.
+
+    Returns
+    -------
+    out : : dpnp.ndarray
+        Boolean array of same shape as `x`.
+
+    See Also
+    --------
+    :obj:`dpnp.iscomplex` : Returns a bool array, where ``True`` if input
+                            element is complex.
+    :obj:`dpnp.isrealobj` : Return ``True`` if `x` is not a complex type.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> a = np.array([1+1j, 1+0j, 4.5, 3, 2, 2j])
+    >>> np.isreal(a)
+    array([False,  True,  True,  True,  True, False])
+
+    """
+    return dpnp.imag(x) == 0
```

Comment:
`dpnp,imag` allocates extra memory for result array, wouldn't it be more efficient to do the same like for `iscomplex`:
```python
    dpnp.check_supported_arrays_type(x)
    if dpnp.issubdtype(x.dtype, dpnp.complexfloating):
        return x.imag == 0
    return dpnp.ones_like(x, dtype=dpnp.bool)
```